fdbrpc/FlowTransport.actor.cpp

```diff
@@ -356,6 +384,11 @@ ACTOR Future<Void> connectionMonitor( Reference<Peer> peer ) {
 					timeouts++;
 				}
 				when (wait( reply.getFuture() )) {
+					double pingLatency = now() - startTime;
+					peer->minPingLatency = std::min(peer->minPingLatency, pingLatency);
+					peer->maxPingLatency = std::max(peer->maxPingLatency, pingLatency);
+					peer->totalPingLatency += pingLatency;
+					peer->totalPingCount++;
```

Comment:
What are your thoughts on using something like `ContinuousSample` for this? That could let us better estimate median and tail latencies, which I think would be preferable to @apkar over mean and max.

It costs slightly more in that you have to compute a random number and add items to a vector, and you then also have to store each of the ping durations over your interval (well, unless you hit the cap).

Also, this isn't capturing any of the pings that end in the other `wait`, should we do something with those?

Reply:
I guess you probably wouldn't end up computing the random number if there are insufficient ping quantities to hit our size cap.

Reply:
I think the max and min is probably good enough, and we do not have to worry about the memory allocation of a large number of samples for each network connection

Reply:
I think having other stats (maybe in particular something like median and one tail), especially for a long running window, might prove useful. We should more carefully consider whether the costs of doing so are worth it.

If I understand correctly, we do something like 1 ping per second on a connection, and we would be logging a connection every `num_connections` seconds. That gives us `num_connections^2` samples. I think if wanted to use 1 MB on a process, this would support ~350 connections.

If we choose our tail latency appropriately (p90?), we can probably get away with capping our number of samples lower, maybe 100 or so? At that point, we would be spending 800 bytes per connection, which is probably reasonable for the number of connections we're holding.

We might also want to consider putting a minimum window on our collection period in this model so that a process with 5 connections can get a non-trivial number of samples between logging.